### ISSUE TITLE
Rename sysroot directories to be more clear

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#101961]
These are effectively user facing because they're used for `rustup link`. So far this has worked reasonably well, but after https://github.com/rust-lang/rust/issues/101691 `stage0-sysroot` will become user facing as well, at which point it will become extremely confusing that there's both that and `stage0`.

I suggest renaming `stage0` to `bootstrap-sysroot`, keeping stage0-sysroot as is and renaming stage1 to stage1-sysroot.

cc @bjorn3 @mark-simulacrum

_Originally posted by @jyn514 in https://github.com/rust-lang/rust/issues/101691#issuecomment-1243988456_

**Comment by jyn514** [comment]
Mentoring instructions: change the names in https://github.com/rust-lang/rust/blob/405f0c21608cb41aabc512eed73b2ff707391cab/src/bootstrap/compile.rs#L1107-L1111

**Comment by andrewpollack** [comment]
What's the status of availability on this? From https://github.com/rust-lang/rust/pull/103286 , this looks both assigned and a PR underway

**Comment by jyn514** [comment]
this is mainly waiting on blog post: https://github.com/rust-lang/rust/pull/103286#issuecomment-1321262082

**Comment by KittyBorgX** [comment]
Can I take this up if this is still up? (I'm new to contributing to the rust repo)

**Comment by jyn514** [comment]
Hmm, I'm no longer sure I want to do this right now. The changes I proposed in https://jyn.dev/2023/01/12/Bootstrapping-Rust-in-2023.html would change the directories names to something else than what I put in the issue here, but the new names (bootstrap-sysroot, dev-sysroot, dist-sysroot) don't make sense until the new `--sysroot` flag is added.

**Comment by KittyBorgX** [comment]
I guess this issue is no longer take-able since the S-blocked label has been added then? (I'm really sorry, I'm new to this :D)

**Comment by jyn514** [comment]
@KittyBorgX yes, sorry - I don't think this is a good issue to work on just now, and adding `--sysroot` is kind of a contentious change that isn't a good first issue.

If you're looking for something to get started with, https://github.com/rust-lang/rust/issues/101691 might be a good issue :) if that's not a good fit, let's continue the discussion on https://rust-lang.zulipchat.com/#narrow/stream/326414-t-infra.2Fbootstrap so it doesn't create noise here.
